src/uwtools/config/core.py

```diff
@@ -675,14 +676,35 @@ def compare_configs(
     :return: False if config files had differences, otherwise True.
     """
 
-    cfg_a = _cli_name_to_config(config_a_format)(config_a_path)
-    cfg_b = _cli_name_to_config(config_b_format)(config_b_path)
+    cfg_a = format_to_config(config_a_format)(config_a_path)
+    cfg_b = format_to_config(config_b_format)(config_b_path)
     logging.info("- %s", config_a_path)
     logging.info("+ %s", config_b_path)
     logging.info("-" * MSGWIDTH)
     return cfg_a.compare_config(cfg_b.data)
 
 
+def format_to_config(cli_name: str) -> Type[Config]:
```

Comment:
This function turns out to be useful outside of just CLI context, and so is renamed. It also now leverages the `FORMAT` object to map strings (e.g. `"nml"`) to their respective config classes (e.g. `NMLConfig`).

Reply:
Nice.